plugins/pulp_rpm/plugins/distributors/rsync/publish.py

```diff
@@ -0,0 +1,132 @@
+import os, functools
+from pulp.plugins.rsync.publish import Publisher, Lazy, RSyncPublishStep
+from pulp_rpm.plugins.distributors.yum import configuration as yum_config
+from pulp_rpm.common import ids
+from pulp.plugins.util.publish_step import RSyncFastForwardUnitPublishStep
+from pulp.server.db.model import Distributor
+
+
+class RPMRsyncPublisher(Publisher):
+
+    def _get_predistributor(self):
+        """
+        Returns the distributor that is configured as predistributor.
+        """
+        predistributor = self.get_config().flatten().get("predistributor_id", None)
+        if predistributor:
+            return Distributor.objects.get_or_404(repo_id=self.repo.id,
+                                                  distributor_id=predistributor)
+        else:
+            raise RuntimeError('RPMRsyncDistributor requires a predistributor to be configured.')
+
+    def _get_extra_path(self):
+        """
+        Returns the relative RPM path that should be appended to
+        """
+        return self.predistributor["config"].get("relative_rpm_path", "")
+
+    def _get_root_publish_dir(self):
+        """
+        Returns the publish directory path for the predistributor
+        :return:
+        """
+        if self.predistributor["config"].get("https", False):
+            return yum_config.get_https_publish_dir(self.get_config())
+        else:
+            return yum_config.get_http_publish_dir(self.get_config())
+
+    def get_repo_content_types(self):
+        """
+        Returns list of file content types supported by this distributor
+        """
+        return [ids.TYPE_ID_RPM, ids.TYPE_ID_DRPM, ids.TYPE_ID_SRPM]
```

Comment:
This could be a tuple that's just an attribute of the class, instead of a function.